src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -120,113 +120,55 @@ public Guid(int a, short b, short c, byte d, byte e, byte f, byte g, byte h, byt
             _k = k;
         }
 
-        [Flags]
-        private enum GuidStyles
-        {
-            None = 0x00000000,
-            AllowParenthesis = 0x00000001, //Allow the guid to be enclosed in parens
-            AllowBraces = 0x00000002, //Allow the guid to be enclosed in braces
-            AllowDashes = 0x00000004, //Allow the guid to contain dash group separators
-            AllowHexPrefix = 0x00000008, //Allow the guid to contain {0xdd,0xdd}
-            RequireParenthesis = 0x00000010, //Require the guid to be enclosed in parens
-            RequireBraces = 0x00000020, //Require the guid to be enclosed in braces
-            RequireDashes = 0x00000040, //Require the guid to contain dash group separators
-            RequireHexPrefix = 0x00000080, //Require the guid to contain {0xdd,0xdd}
-
-            HexFormat = RequireBraces | RequireHexPrefix,                      /* X */
-            NumberFormat = None,                                                  /* N */
-            DigitFormat = RequireDashes,                                         /* D */
-            BraceFormat = RequireBraces | RequireDashes,                         /* B */
-            ParenthesisFormat = RequireParenthesis | RequireDashes,                    /* P */
-
-            Any = AllowParenthesis | AllowBraces | AllowDashes | AllowHexPrefix,
-        }
-        private enum GuidParseThrowStyle
+        private enum GuidParseThrowStyle : byte
         {
             None = 0,
             All = 1,
             AllButOverflow = 2
         }
-        private enum ParseFailureKind
-        {
-            None = 0,
-            ArgumentNull = 1,
-            Format = 2,
-            FormatWithParameter = 3,
-            NativeException = 4,
-            FormatWithInnerException = 5
-        }
 
         // This will store the result of the parsing.  And it will eventually be used to construct a Guid instance.
         private struct GuidResult
         {
+            internal readonly GuidParseThrowStyle _throwStyle;
             internal Guid _parsedGuid;
-            internal GuidParseThrowStyle _throwStyle;
 
-            private ParseFailureKind _failure;
+            private bool _overflow;
             private string _failureMessageID;
             private object _failureMessageFormatArgument;
-            private string _failureArgumentName;
-            private Exception _innerException;
 
-            internal void Init(GuidParseThrowStyle canThrow)
+            public GuidResult(GuidParseThrowStyle canThrow) : this()
```

Comment:
Nit: The GuidResult constructor is public, but all other GuidResult methods are internal.

Reply:
Ok, will fix.